### PR TITLE
 Adds DefaultCacheStorage#listDigests.

### PR DESCRIPTION
**Pull request by coollog**
Part of #637 

`DefaultCacheStorageReader` - the logic behind `DefaultCacheStorage#listDigests/retrieve/select`

### CODE REVIEW

**Comment by loosebazooka** [comment]
Assuming this needs to be rebased?

File: jib-core/src/main/java/com/google/cloud/tools/jib/ncache/DefaultCacheStorageReader.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    try (Stream<Path> layerDirectories =
+        Files.list(defaultCacheStorageFiles.getLayersDirectory())) {
+      List<Path> layerDirectoriesList = layerDirectories.collect(Collectors.toList());
+      List<DescriptorDigest> layerDigests = new ArrayList<>(layerDirectoriesList.size());
+      for (Path layerDirectory : layerDirectoriesList) {
+        try {
+          layerDigests.add(DescriptorDigest.fromHash(layerDirectory.getFileName().toString()));
+
+        } catch (DigestException ex) {
+          throw new CacheCorruptedException("Found non-digest file in layers directory", ex);
```
Comment (loosebazooka) [inline jib-core/src/main/java/com/google/cloud/tools/jib/ncache/DefaultCacheStorageReader.java]: What's the solution when this happens? Delete the layer directory?

Comment (loosebazooka) [reply jib-core/src/main/java/com/google/cloud/tools/jib/ncache/DefaultCacheStorageReader.java]: Do we plan on having a cacheFixer?

Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/ncache/DefaultCacheStorageReader.java]: Yea, for now we'll just defer to the user of the class to handle corruptions like this.

**Comment by coollog** [comment]
@loosebazooka Reading won't be corrupted by a writer is "guaranteed" by writes always writing atomically (writing to a temporary file and atomically moving to the actual location).

**Pull request merged**